tests/integration/kubernetes/gha-run.sh

```diff
@@ -241,13 +242,90 @@ function cleanup() {
+	case ${SNAPSHOTTER:-} in
+		nydus) cleanup_nydus_snapshotter ;;
+		*) >&2 echo "${SNAPSHOTTER:-} flavour is not supported"; exit 2 ;;
+	esac
+	echo "::endgroup::"
+}
+
+function deploy_nydus_snapshotter() {
+    echo "::group::deploy_nydus_snapshotter"
+	ensure_yq
+
+	local nydus_snapshotter_install_dir="/tmp/nydus-snapshotter"
+	if [ -d "${nydus_snapshotter_install_dir}" ]; then
+		rm -rf "${nydus_snapshotter_install_dir}"
+	fi
+	mkdir -p "${nydus_snapshotter_install_dir}"
+	nydus_snapshotter_url=$(get_from_kata_deps "externals.nydus-snapshotter.url")
+	nydus_snapshotter_version=$(get_from_kata_deps "externals.nydus-snapshotter.version")
+	git clone -b "${nydus_snapshotter_version}" "${nydus_snapshotter_url}" "${nydus_snapshotter_install_dir}"
+
+	pushd "$nydus_snapshotter_install_dir"
+	if [ "${PULL_TYPE}" == "guest-pull" ]; then
+		# Enable guest pull feature in nydus snapshotter
+		yq write -i misc/snapshotter/base/nydus-snapshotter.yaml 'data.FS_DRIVER' "proxy" --style=double
+	else
+		>&2 echo "Invalid pull type"; exit 2 
+	fi
+	
+	# Disable to read snapshotter config from configmap
+	yq write -i misc/snapshotter/base/nydus-snapshotter.yaml 'data.ENABLE_CONFIG_FROM_VOLUME' "false" --style=double
+	# Enable to run snapshotter as a systemd service
+	yq write -i misc/snapshotter/base/nydus-snapshotter.yaml 'data.ENABLE_SYSTEMD_SERVICE' "true" --style=double
+	# Enable "runtime specific snapshotter" feature in containerd when configuring containerd for snapshotter
+	yq write -i misc/snapshotter/base/nydus-snapshotter.yaml 'data.ENABLE_RUNTIME_SPECIFIC_SNAPSHOTTER' "true" --style=double
+
+	# Deploy nydus snapshotter as a daemonset
+	kubectl create -f "misc/snapshotter/nydus-snapshotter-rbac.yaml"
+	if [ "${KUBERNETES}" = "k3s" ]; then
+		kubectl apply -k "misc/snapshotter/overlays/k3s"
+	else
+		kubectl apply -f "misc/snapshotter/base/nydus-snapshotter.yaml"
+	fi
+	popd
+
+	kubectl rollout status daemonset nydus-snapshotter -n nydus-system --timeout ${SNAPSHOTTER_DEPLOY_WAIT_TIMEOUT}
+	
+	echo "::endgroup::"
+	echo "::group::nydus snapshotter logs"
+	pods_name=$(kubectl get pods --selector=app=nydus-snapshotter -n nydus-system -o=jsonpath='{.items[*].metadata.name}')
+	kubectl logs ${pods_name} -n nydus-system
+	kubectl describe pod ${pods_name} -n nydus-system
+	echo "::endgroup::"
+}
+
+function cleanup_nydus_snapshotter() {
+	echo "cleanup_nydus_snapshotter"
+	local nydus_snapshotter_install_dir="/tmp/nydus-snapshotter"
+	if [ ! -d "${nydus_snapshotter_install_dir}" ]; then
+		>&2 echo "nydus snapshotter dir not found"
+		exit 1
+	fi
+	
+	pushd "$nydus_snapshotter_install_dir"
+
+	if [ "${KUBERNETES}" = "k3s" ]; then
+		kubectl delete -k "misc/snapshotter/overlays/k3s"
+	else
+		kubectl delete -f "misc/snapshotter/base/nydus-snapshotter.yaml"
+	fi
+	sleep 180s
+	kubectl delete -f "misc/snapshotter/nydus-snapshotter-rbac.yaml"
+	popd
+	sleep 180s
```

Comment:
rbac deletion usually is quite fast, I'd drop this to 30s or so, instead of 3 minutes.